ports/psoc6/modules/machine/machine_rtc.c

```diff
@@ -60,62 +69,42 @@ typedef struct _machine_rtc_obj_t {
 // singleton RTC object
 STATIC const machine_rtc_obj_t machine_rtc_obj = {{&machine_rtc_type}};
 
-
+/* This function is run from main.c to init the RTC at boot time. This will set the RTC to PSoC default time: 1st Jan 2000*/
 void rtc_init(void) {
     cy_rslt_t result = cyhal_rtc_init(&psoc6_rtc);
 
     if (CY_RSLT_SUCCESS != result) {
         mp_raise_ValueError(MP_ERROR_TEXT("cyhal_rtc_init failed !"));
     }
-
-    struct tm current_date_time = { .tm_mon = 1, .tm_mday = 1 };
-    result = cyhal_rtc_write(&psoc6_rtc, &current_date_time);
-
-    if (CY_RSLT_SUCCESS != result) {
-        mp_raise_ValueError(MP_ERROR_TEXT("cyhal_rtc_write failed !"));
-    }
-}
-
-
-void rtc_deinit(void) {
-
 }
 
-
+// Machine RTC methods - port-specific definitions
+// RTC constructor
 STATIC mp_obj_t machine_rtc_make_new(const mp_obj_type_t *type, size_t n_args, size_t n_kw, const mp_obj_t *args) {
     mp_arg_check_num(n_args, n_kw, 0, 0, false);
-    bool r = cyhal_rtc_is_enabled(&psoc6_rtc);
-
-    if (!r) {
-        // This shouldn't happen as rtc_init() is already called in main so
-        // it's here just in case
-        rtc_init();
-    }
-
     // return constant object
     return (mp_obj_t)&machine_rtc_obj;
 }
 
-STATIC mp_obj_t machine_rtc_datetime(mp_uint_t n_args, const mp_obj_t *args) {
+// Helper function to set/get datetime
+STATIC mp_obj_t machine_rtc_datetime_helper(mp_uint_t n_args, const mp_obj_t *args) {
     if (n_args == 1) {
         struct tm current_date_time = {0};
         cy_rslt_t result = cyhal_rtc_read(&psoc6_rtc, &current_date_time);
-
         if (CY_RSLT_SUCCESS != result) {
             mp_raise_ValueError(MP_ERROR_TEXT("cyhal_rtc_read failed !"));
         }
 
         mp_obj_t tuple[8] = {
-            mp_obj_new_int(current_date_time.tm_year),
-            mp_obj_new_int(current_date_time.tm_mon),
+            mp_obj_new_int(current_date_time.tm_year + TM_YEAR_BASE),
```

Comment:
To comply with our HAL. struct tm should hold years since 1900(= TM_YEAR_BASE). I did not go into the algorithm details of how this reference is effecting but this struct passed to HAL seems to expect this subtraction and addition. Reference example taken from here: https://infineon.github.io/psoc6hal/html/group__group__hal__rtc.html

Reply:
So why is 2015 used instead of 1900 ?

Reply:
Ok, that is bcz of differences in mpy and hal. Let me explain:
1. RTC.deinit() as mentioned in mpy docs (https://docs.micropython.org/en/latest/library/machine.RTC.html), expects to reset RTC to 1st Jan'2015 and hence this year.
2. As a user the flow should be the following:
a) Create the RTC object --> This will do nothing but create an object (no initializations). If you check the year at this time, it would be 1st Jan'2000.
b) Initialize the RTC --> Here you should pass the starting date and time to be set for your application.
c) Next whenever you try to get the current time using now() or datetime(), the secs/minutes/hours elapsed will then be calculated from the time you set during init.  

All this is done to comply with what mpy expects and what our HAL does(to which we cannot alter much). But the good thing is there are APIs that allow user to keep track of this date and time and we have added this in the documentation as well. Does this answer?